### PR TITLE
Overmap scent traces

### PR DESCRIPTION
**Pull request by kevingranade**
This adds support for the player leaving scent traces on the overmap scale, in the future hordes will be able to track the player via these.

### CODE REVIEW

**Comment by Zireael07** [comment]
Great!

File: src/overmap.cpp [hunk 1]
```diff
@@ -1720,6 +1762,7 @@ void overmap::draw(WINDOW *w, WINDOW *wbar, const tripoint &center,
                 ter_sym   = '@';
             } else if( data.debug_weather && get_weather_glyph( point( omx, omy ), ter_color, ter_sym ) ) {
                 // ter_color and ter_sym have been set by get_weather_glyph
+            } else if( data.debug_scent && get_scent_glyph( cur_pos, ter_color, ter_sym ) ) {
```
Comment (Coolthulhu) [inline src/overmap.cpp]: This line (I think?) makes gcc complain later on:
```
src/overmap.cpp: In static member function ‘static void overmap::draw(WINDOW*, WINDOW*, const tripoint&, const tripoint&, bool, bool, input_context*, const overmap::draw_data_t&)’:
src/overmap.cpp:1861:40: error: ‘ter_sym’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
                     if( ter_sym == '+' || ter_sym == '-' ) {
                                        ^
src/overmap.cpp:1898:54: error: ‘ter_color’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
                 mvwputch(w, j, i, ter_color, ter_sym);
                                                      ^
```

Comment (Coolthulhu) [inline src/overmap.cpp]: After fixing that by giving a default value to `ter_sym`, same thing happened with `ter_color`.

**Comment by Coolthulhu** [comment]
The functionality seems to work, though the scent doesn't built up - it immediately goes to full value.
Not sure why does gcc complain now and it didn't before.

**Comment by kevingranade** [comment]
Scent strength changing over time is future functionality, and will be based on expected times for the scent to last, so I won't be able to balance it until I have monsters paying attention to it.

**Comment by kevingranade** [comment]
That should handle the warning, are you insisting it needs progressive scent strength for landing, or was that just an observation?

**Comment by Coolthulhu** [comment]
Just an observation. I expected the numbers to increase from 1 to 9, but got just 3 and I was wondering what's up.

**Comment by kevingranade** [comment]
I just hacked in the debug view to make sure it works, I didn't normalize
against the actual range of values.

**Pull request merged**